spec/API_specification/array_api/fourier_transform_functions.py

```diff
@@ -0,0 +1,442 @@
+    """
+    Computes the one-dimensional inverse discrete Fourier transform.
+
+    Parameters
+    ----------
+    x: array
+        input array. Should have a floating-point data type.
+    n: int
+        length of the transformed axis of the output. If
+
+        - ``n`` is larger than the length of the input array, the input array must be zero-padded.
+        - ``n`` is less than the length of the input array, the input array must be trimmed to length ``n``.
+
+        if not provided, the length of the transformed axis of the output must equal the length of the input along the axis specified by ``axis``. Default: ``None``.
+    axis: int
+        axis (dimension) over which to compute the inverse Fourier transform. If set to ``-1``, the function must compute the Fourier transform over the last axis (dimension). Default: ``-1``.
+    norm: Literal['backward', 'ortho', 'forward']
+        normalization mode. Should be one of the following modes:
+
+        - ``'backward'``: normalize by ``1/n``.
+        - ``'ortho'``: normalize by ``1/sqrt(n)``
+        - ``'forward'``: no normalization.
+
+        Default: ``'backward'``.
+
+    Returns
+    -------
+    out: array
+        an array transformed along the axis (dimension) indicated by ``axis``. The returned array must have a complex floating-point data type determined by :ref:`type-promotion`.
+    """
+
+
+def fftn(x: array, /, *, s: Union[Sequence[int], Tuple[int, ...]] = None, axes: Union[Sequence[int], Tuple[int, ...]] = None, norm: Literal['backward', 'ortho', 'forward'] = 'backward') -> array:
+    """
+    Computes the n-dimensional discrete Fourier transform.
+
+
+    .. note::
+       Applying the n-dimensional inverse discrete Fourier transform to the output of this function must return the original (i.e., non-transformed) input array within numerical accuracy: ``ifftn(fftn(x)) == x``.
+
+    Parameters
+    ----------
+    x: array
+        input array. Should have a floating-point data type.
+    n: int
+        length of the transformed axis of the output. If
+
+        - ``n`` is larger than the length of the input array, the input array must be zero-padded.
+        - ``n`` is less than the length of the input array, the input array must be trimmed to length ``n``.
+
+        if not provided, the length of the transformed axis of the output must equal the length of the input along the axis specified by ``axis``. Default: ``None``.
+    s: Union[Sequence[int], Tuple[int, ...]]
+        size of each transformed axis of the output. If
+
+        - ``s`` is larger than the length of the input array, each axis ``i`` of the input array must be zero-padded.
+        - ``s`` is less than the length of the input array, each axis ``i`` of the input array must be trimmed to the length ``s[i]``.
+
+        if not provided, the length of the transformed axes of the output must equal the length of the input along the axes specified by ``axes``. Default: ``None``.
```

Comment:
```suggestion
        If ``axes`` is ``None``, the last ``len(s)`` axes must be transformed, and, thus, ``j`` represents an axis index computed according to ``N - len(s) + i``, where ``N`` is the number of axes (e.g., if ``N = 4`` and ``s = (256, 256)``, then ``i = 0`` corresponds to axis ``2`` and ``i = 1`` corresponds to axis ``3``, the last axis).
        
        If ``axes`` is not ``None``, size ``s[i]`` corresponds to the size of axis ``axes[j]`` where ``i == j``.
 
        If not provided, the size of each transformed axis (dimension) in the output array must equal the size of the corresponding axis in the input array.
        
        Default: ``None``.
```

Reply:
PyTorch allows `s[i]` to equal `-1` (see https://pytorch.org/docs/stable/generated/torch.fft.fftn.html). This is explicitly spelled out in the Torch docs. From looking at NumPy, I don't believe NumPy allows something similar. Allowing `-1` allows doing something like

```python
fftn(x, s=(-1,256))
```

rather than

```python
sh = x.shape
fftn(x, s=(sh[x.ndim-2], 256))
```

which, at first blush, is convenient.

Not sure if we want to follow suit in the here.

Reply:
I think the resolution for this issue (#476) is yes.